ArtOfIllusion/src/artofillusion/image/ExternalImage.java

```diff
@@ -0,0 +1,479 @@
+/* Copyright (C) 2017 by Petri Ihalainen
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.image;
+
+import artofillusion.*;
+import artofillusion.math.*;
+import buoy.widget.BStandardDialog;
+import java.awt.*;
+import java.awt.image.*;
+import java.io.*;
+import java.util.Date;
+import javax.imageio.*;
+
+public class ExternalImage extends ImageMap
+{
+  private File externalFile;
+  private String loadingError, imageType;
+  private ImageMap imageMap;
```

Comment:
I see your point. Still not my favorite design, but I do understand the logic. I'll try to get this merged in the next couple of days. I want to get some of the commented-out code snips removed for real.